src/ScheduleFunctions.cpp

```diff
@@ -461,17 +462,22 @@ Stmt build_produce(Function f) {
                     buf_name += ".buffer";
                     Expr buffer = Variable::make(type_of<struct buffer_t *>(), buf_name);
                     extern_call_args.push_back(buffer);
+                    buffer_names.push_back(buf_name);
                 }
             } else if (arg.is_buffer()) {
                 BufferPtr b = arg.buffer;
                 Parameter p(b.type(), true, b.dimensions(), b.name());
                 p.set_buffer(b);
-                Expr buf = Variable::make(type_of<struct buffer_t *>(), b.name() + ".buffer", p);
+                string buf_name = b.name() + ".buffer";
+                Expr buf = Variable::make(type_of<struct buffer_t *>(), buf_name, p);
                 extern_call_args.push_back(buf);
+                buffer_names.push_back(buf_name);
             } else if (arg.is_image_param()) {
                 Parameter p = arg.image_param;
-                Expr buf = Variable::make(type_of<struct buffer_t *>(), p.name() + ".buffer", p);
+                string buf_name = p.name() + ".buffer";
+                Expr buf = Variable::make(type_of<struct buffer_t *>(), buf_name, p);
                 extern_call_args.push_back(buf);
+                buffer_names.push_back(buf_name);
```

Comment:
Good call, though I think the buffer_t itself in this case also doesn't need annotation -- it's allocated and filled in by the caller.